components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/claims/BFSIDefaultOIDCClaimsCallbackHandler.java

```diff
@@ -0,0 +1,169 @@
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+import org.wso2.carbon.identity.openidconnect.DefaultOIDCClaimsCallbackHandler;
+
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * This call back handler adds ob specific additional claims to self contained JWT access token.
+ */
+public class BFSIDefaultOIDCClaimsCallbackHandler extends DefaultOIDCClaimsCallbackHandler {
+
+    private static Log log = LogFactory.getLog(BFSIDefaultOIDCClaimsCallbackHandler.class);
+    Map<String, Object> identityConfigurations = IdentityExtensionsDataHolder.getInstance().getConfigurationMap();
+
+    @Override
+    public JWTClaimsSet handleCustomClaims(JWTClaimsSet.Builder jwtClaimsSetBuilder, OAuthTokenReqMessageContext
+            tokenReqMessageContext) throws IdentityOAuth2Exception {
+
+        /*  accessToken property check is done to omit the following claims getting bound to id_token
+             The access token property is added to the ID token message context before this method is invoked. */
+        try {
+            if (CommonUtils.isRegulatoryApp(tokenReqMessageContext.getOauth2AccessTokenReqDTO()
+                    .getClientId()) && (tokenReqMessageContext.getProperty("accessToken") == null)) {
+
+                Map<String, Object> userClaimsInOIDCDialect = new HashMap<>();
+                JWTClaimsSet jwtClaimsSet = getJwtClaimsFromSuperClass(jwtClaimsSetBuilder, tokenReqMessageContext);
+                if (jwtClaimsSet != null) {
+                    userClaimsInOIDCDialect.putAll(jwtClaimsSet.getClaims());
+                }
+                addCnfClaimToOIDCDialect(tokenReqMessageContext, userClaimsInOIDCDialect);
+                addConsentIDClaimToOIDCDialect(tokenReqMessageContext, userClaimsInOIDCDialect);
+                updateSubClaim(tokenReqMessageContext, userClaimsInOIDCDialect);
+
+                for (Map.Entry<String, Object> claimEntry : userClaimsInOIDCDialect.entrySet()) {
+                    if (IdentityCommonConstants.SCOPE.equals(claimEntry.getKey())) {
+                        String[] nonInternalScopes = IdentityCommonUtils
+                                .removeInternalScopes(claimEntry.getValue().toString()
+                                        .split(IdentityCommonConstants.SPACE_SEPARATOR));
+                        jwtClaimsSetBuilder.claim(IdentityCommonConstants.SCOPE, StringUtils.join(nonInternalScopes,
+                                IdentityCommonConstants.SPACE_SEPARATOR));
+                    } else {
+                        jwtClaimsSetBuilder.claim(claimEntry.getKey(), claimEntry.getValue());
+                    }
+                }
+                return jwtClaimsSetBuilder.build();
+            }
+        } catch (RequestObjectException e) {
+            throw new IdentityOAuth2Exception(e.getMessage(), e);
```

Comment:
Let's add an error log here.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/884f24f4261bc92a886a68a4e8a5bf56e7a61136